plan.go

```diff
@@ -136,44 +137,29 @@ func extractSelectionSet(ctx *PlanningContext, insertionPoint []string, parentTy
 				childrenStepsResult = append(childrenStepsResult, steps...)
 				continue
 			}
-			if loc == location {
-				if selection.SelectionSet == nil {
-					selectionSetResult = append(selectionSetResult, selection)
-				} else {
-					newField := *selection
-					selectionSet, childrenSteps, err := extractSelectionSet(
-						ctx,
-						append(insertionPoint, selection.Alias),
-						selection.Definition.Type.Name(),
-						selection.SelectionSet,
-						location,
-						childstep,
-					)
-					if err != nil {
-						return nil, nil, err
-					}
-					newField.SelectionSet = selectionSet
-					selectionSetResult = append(selectionSetResult, &newField)
-					childrenStepsResult = append(childrenStepsResult, childrenSteps...)
-				}
+			if loc != location {
+				// field transitions to another service location
+				remoteSelections = append(remoteSelections, selection)
```

Comment:
This is the only actual change to this conditional: transitions in service location now collect remote fields into an array and resolve them all at once later in the process, versus resolving each remote field individually here. There's a lot less churn resolving them together, and it avoids the situation that `mergedWithExistingStep` was previously trying to address.